src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcImportProjectFeature.cs

```diff
@@ -31,54 +31,59 @@ public IReadOnlyList<RazorSourceDocument> GetImports(RazorProjectItem projectIte
         }
 
         // Internal for testing
-        internal void AddHierarchicalImports(RazorProjectItem projectItem, List<RazorSourceDocument> imports)
+        internal void AddHierarchicalImports(RazorProjectItem projectItem, List<RazorProjectItem> imports)
         {
             // We want items in descending order. FindHierarchicalItems returns items in ascending order.
             var importProjectItems = ProjectEngine.FileSystem.FindHierarchicalItems(projectItem.FilePath, ImportsFileName).Reverse();
-            foreach (var importProjectItem in importProjectItems)
+            imports.AddRange(importProjectItems);
+        }
+
+        private class DefaultDirectivesProjectItem : RazorProjectItem
+        {
+            private readonly byte[] _defaultImportBytes;
+
+            private DefaultDirectivesProjectItem()
             {
-                RazorSourceDocument importSourceDocument;
-
-                if (importProjectItem.Exists)
-                {
-                    importSourceDocument = RazorSourceDocument.ReadFrom(importProjectItem);
-                }
-                else
-                {
-                    // File doesn't exist on disk so just add a marker source document as an identifier for "there could be something here".
-                    var sourceDocumentProperties = new RazorSourceDocumentProperties(importProjectItem.FilePath, importProjectItem.RelativePhysicalPath);
-                    importSourceDocument = RazorSourceDocument.Create(string.Empty, sourceDocumentProperties);
-                }
-
-                imports.Add(importSourceDocument);
+                var preamble = Encoding.UTF8.GetPreamble();
+                var content = @"
+@using System
+@using System.Collections.Generic
+@using System.Linq
+@using System.Threading.Tasks
+@using Microsoft.AspNetCore.Mvc
+@using Microsoft.AspNetCore.Mvc.Rendering
+@using Microsoft.AspNetCore.Mvc.ViewFeatures
+@inject global::Microsoft.AspNetCore.Mvc.Rendering.IHtmlHelper<TModel> Html
+@inject global::Microsoft.AspNetCore.Mvc.Rendering.IJsonHelper Json
+@inject global::Microsoft.AspNetCore.Mvc.IViewComponentHelper Component
+@inject global::Microsoft.AspNetCore.Mvc.IUrlHelper Url
+@inject global::Microsoft.AspNetCore.Mvc.ViewFeatures.IModelExpressionProvider ModelExpressionProvider
+@addTagHelper Microsoft.AspNetCore.Mvc.Razor.TagHelpers.UrlResolutionTagHelper, Microsoft.AspNetCore.Mvc.Razor
+";
+                var contentBytes = Encoding.UTF8.GetBytes(content);
+
+                _defaultImportBytes = new byte[preamble.Length + contentBytes.Length];
+                preamble.CopyTo(_defaultImportBytes, 0);
+                contentBytes.CopyTo(_defaultImportBytes, preamble.Length);
             }
+
+            public override string BasePath => null;
+
+            public override string FilePath => null;
+
+            public override string PhysicalPath => null;
+
+            public override bool Exists => false;
```

Comment:
Should be true YO

Reply:
Solid catch, should be that way in the non-1_X as well 👍